File: ruby/src/sorting/stupid_sort.rb
require_relative '../common/array_utils'

module Sorting
  class << self
    def stupid_sort(array, &comparator)
      (0..array.size - 1).each do |i|
        (i + 1..array.size - 1).each do |j|
          ArrayUtils.swap(array, i, j) if (yield(array[i], array[j]) > 0)
        end
      end

      array
    end
  end
end

File: java/src/sorting/StupidSort.java
package sorting;
   
import java.util.Comparator;

public class StupidSort {
    public static void swap(int[] array, int foo, int bar) {
	int temp = array[foo];
	array[foo] = array[bar];
	array[bar] = temp;
    }

    public static void stupidSort(int[] array, Comparator<Integer> comparator) {
	for (int i = 0; i < array.length; i++) {
	    for (int j = i + 1; j < array.length; j++) {
		if (comparator.compare(array[i], array[j]) > 0) {
		    swap(array, i, j);
		}
	    } 
	}
    }    
}